Sources/NIOHTTP2/ConnectionStateMachine/FrameReceivingStates/ReceivingHeadersState.swift

```diff
@@ -74,7 +86,10 @@ extension ReceivingHeadersState where Self: LocallyQuiescingState {
         let localSupportsExtendedConnect = self.localSupportsExtendedConnect
         let remoteSupportsExtendedConnect = self.remoteSupportsExtendedConnect
 
-        if streamID.mayBeInitiatedBy(.client) && streamID > self.lastRemoteStreamID {
+        // We are in `LocallyQuiescingState`. The sender of the GOAWAY is `self.role`, so the receiver is the inverse of `self.role`.
+        let receiver = self.role.inverse()
```

Comment:
I find `receiver` really misleading because the state machine is dealing with receiving headers and `receiver` doesn't refer to the state machine but rather the remote peer. I think `remotePeer` would be clearer here.